apps/astarte_pairing/lib/astarte_pairing/queries.ex

```diff
@@ -223,6 +224,34 @@ defmodule Astarte.Pairing.Queries do
     {:ok, count}
   end
 
+  def create_ownership_voucher(realm_name, voucher_blob, pkey_blob) do
+    keyspace_name = Realm.keyspace_name(realm_name)
+    opts = [prefix: keyspace_name, consistency: Consistency.device_info(:write)]
+
+    timestamp_ms =
+      DateTime.utc_now()
+      |> DateTime.add(7 * 24 * 60 * 60, :second)
+      |> DateTime.to_unix(:millisecond)
+
+    %OwnershipVoucher{
+      voucher: voucher_blob,
+      pkey: pkey_blob,
+      exp_timestamp: timestamp_ms
+    }
+    |> Repo.insert(opts)
+  end
+
+  def update_device_registration_voucher(realm_name, device_id, voucher_id) do
+    keyspace_name = Realm.keyspace_name(realm_name)
+
+    %Device{}
+    |> Ecto.Changeset.change(%{
+      device_id: device_id,
+      ownership_voucher: voucher_id
+    })
```

Comment:
it's an insertion, you don't need to change manually


```suggestion
    %Device{
      device_id: device_id,
      ownership_voucher: voucher_id
    }
```